src/global/reco/GeneratedJets_factory.cc

```diff
@@ -1,52 +1,67 @@
 // SPDX-License-Identifier: LGPL-3.0-or-later
 // Copyright (C) 2023 Zhongling Ji, Derek Anderson
 
-#include "GeneratedJets_factory.h"
-
+// standard c includes
 #include <memory>
-
 #include <JANA/JEvent.h>
-#include "services/log/Log_service.h"
-
+#include <services/log/Log_service.h>
+// event data model definitions
 #include <edm4hep/MCParticle.h>
+// factory-specific includes
+#include "GeneratedJets_factory.h"
+
 
 
 namespace eicrecon {
 
     void GeneratedJets_factory::Init() {
 
-        // This prefix will be used for parameters
-        std::string param_prefix = "reco:" + GetTag();
-
-        // Set input data tags properly
-        InitDataTags(param_prefix);
-
         // SpdlogMixin logger initialization, sets m_log
-        InitLogger(GetApplication(), param_prefix, "info");
+        InitLogger(GetApplication(), GetPrefix(), "info");
 
+        // initialize jet reconstruction algorithm
         m_jet_algo.init(m_log);
-    }
 
-    void GeneratedJets_factory::ChangeRun(const std::shared_ptr<const JEvent> &event) {
-        // Nothing to do here
-    }
+    }  // end 'Init()'
+
+
+
+    void GeneratedJets_factory::BeginRun(const std::shared_ptr<const JEvent> &event) {
+
+      // nothing to do here
+
+    }  // end 'BeginRun(std::shared_ptr<JEvent&>)'
+
+
 
     void GeneratedJets_factory::Process(const std::shared_ptr<const JEvent> &event) {
-        auto mc_particles = event->Get<edm4hep::MCParticle>("MCParticles");
 
+        // grab input collection
+        auto input = event->Get<edm4hep::MCParticle>(GetInputTags()[0]);
```

Comment:
Avoid conversion to `std::vector`
```suggestion
        auto input = static_cast<const edm4hep::MCParticleCollection*>(event->GetCollectionBase(GetInputTags()[0]));
```